src/main/java/seedu/address/logic/commands/NumSortCommand.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ATTRIBUTE;
+
+import java.util.Optional;
+
+import seedu.address.model.Model;
+import seedu.address.model.attribute.NumericalValueBasedAttributeComparator;
+import seedu.address.model.person.AttributeBasedPersonComparator;
+
+/**
+ * Sort the candidates with the provided attribute.
+ * Guarantees: immutable.
+ */
+public class NumSortCommand extends SortCommand {
+    public static final String COMMAND_WORD = "sort-num";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sort the candidates in ascending order based on numerical value of the specified attribute.\n"
+            + "Candidates lacking the specified attribute name will be placed last, preserving their original order.\n"
+            + "Among Candidates with the specified attribute name, "
+            + "those without a numerical value will be placed last.\n"
+            + "Parameters: ATTRIBUTE_NAME (case-insensitive)\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ATTRIBUTE + "Graduation Year";
+    public static final String MESSAGE_WARNING_MISSING_NUMERICALS =
+            "WARNING! Only entries up to index %1$d have valid numerical values for the specified attribute.\n";
```

Comment:
Suggestion: how about changing it to "the specified attribute name"? It would be more accurate to the current implementation 👍